django/db/models/expressions.py

```diff
@@ -1290,6 +1290,56 @@ def get_db_converters(self, connection):
         ) + self.target.get_db_converters(connection)
 
 
+class Cols(Expression):
+    def __init__(self, alias, targets, sources, output_field):
+        super().__init__(output_field=output_field)
+        self.alias, self.targets, self.sources = alias, targets, sources
```

Comment:
I like how you managed to combine both `Cols` and `MultiColSource` source 🎉 

I'm not sure `Cols` is good name as it's practically a _pair_. Maybe `ColsPair` would be a better name?

Reply:
WDYT about going with 2 classes (as explained below)?